server/lib/chat/chat-api.ts

```diff
@@ -385,4 +387,26 @@ export class AdminChatApi {
       isAdmin: true,
     })
   }
+
+  @httpDelete('/:channelId/messages/:messageId')
+  @httpBefore(checkAllPermissions('moderateChatChannels'))
+  async deleteMessage(ctx: RouterContext): Promise<void> {
+    const {
+      params: { channelId, messageId },
+    } = validateRequest(ctx, {
+      params: Joi.object<{ channelId: SbChannelId; messageId: string }>({
+        channelId: serialIdSchema().required(),
+        messageId: Joi.string().required(),
+      }),
+    })
+
+    await deleteChannelMessage(messageId)
```

Comment:
Why is this accessing the model directly and publishing its own events? This is the job of the service, it should be written like all the other stuff in this API